backend/src/controllers/RestaurantController.ts

```diff
@@ -1,19 +1,23 @@
-import { Request, Response } from 'express';
+import { Request, Response, NextFunction } from 'express';
 import Restaurant from '../models/restaurant';
+import logger from '../utils/logger';
 
 //http://localhost:3000/restaurants/search?searchQuery=pizza&selectedCuisines=Italian,Mexican&sortOption=restaurantName&page=2
-const searchRestaurants = async (req: Request, res: Response) => {
+const searchRestaurants = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Potential unused parameter in function signature</b></div>
<div id="fix">

Consider adding error handling for the `next` parameter. It might be beneficial to use this parameter to pass errors to the Express error handling middleware.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -6,7 +6,7 @@
 -const searchRestaurants = async (req: Request, res: Response, next: NextFunction) => {
 +const searchRestaurants = async (req: Request, res: Response, next: NextFunction) => {
    try {
      // ... existing code ...
    } catch (error) {
 -    // Handle error
 +    next(error);
    }
  }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/Food-Ordering-App/pull/47#issuecomment-2439370000>#eef1bf</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged